src/precice/tests/ParallelTests.cpp

```diff
@@ -1104,6 +1105,936 @@ BOOST_AUTO_TEST_CASE(UserDefinedMPICommunicatorPetRBF)
+      int vF = cplInterface.setMeshVertex(meshID, coordTwoF.data());
+
+      int eAB = cplInterface.setMeshEdge(meshID, vA, vB);
+      int eBC = cplInterface.setMeshEdge(meshID, vB, vC);
+      int eAC = cplInterface.setMeshEdge(meshID, vA, vC);
+      int eCD = cplInterface.setMeshEdge(meshID, vC, vD);
+      int eAD = cplInterface.setMeshEdge(meshID, vA, vD);
+      int eBF = cplInterface.setMeshEdge(meshID, vB, vF);
+      int eFC = cplInterface.setMeshEdge(meshID, vF, vC);
+
+      cplInterface.setMeshTriangle(meshID, eAB, eBC, eAC);
+      cplInterface.setMeshTriangle(meshID, eAD, eAC, eCD);
+      cplInterface.setMeshTriangle(meshID, eBC, eFC, eBF);
+
+      double maxDt = cplInterface.initialize();
+
+      double valueA, valueB, valueC, valueD, valueF;
+      cplInterface.readScalarData(dataID, vA, valueA);
+      cplInterface.readScalarData(dataID, vB, valueB);
+      cplInterface.readScalarData(dataID, vC, valueC);
+      cplInterface.readScalarData(dataID, vD, valueD);
+      cplInterface.readScalarData(dataID, vF, valueF);
+
+      double localCalculatedIntegral = (math::geometry::triangleArea(coordTwoA, coordTwoB, coordTwoC) * (valueA + valueB + valueC) +
+                                        math::geometry::triangleArea(coordTwoA, coordTwoC, coordTwoD) * (valueA + valueC + valueD) +
+                                        math::geometry::triangleArea(coordTwoB, coordTwoC, coordTwoF) * (valueB + valueC + valueF)) /
+                                       3.0;
+      utils::MasterSlave::allreduceSum(localCalculatedIntegral, calculatedIntegral, 1);
+
+      cplInterface.advance(maxDt);
+      BOOST_TEST(!cplInterface.isCouplingOngoing(), "Receiving participant should have to advance once!");
+      cplInterface.finalize();
+
+    } else {
+        int vB = cplInterface.setMeshVertex(meshID, coordTwoB.data());
+        int vE = cplInterface.setMeshVertex(meshID, coordTwoE.data());
+        int vF = cplInterface.setMeshVertex(meshID, coordTwoF.data());
+
+        int eBE = cplInterface.setMeshEdge(meshID, vB, vE);
+        int eBF = cplInterface.setMeshEdge(meshID, vB, vF);
+        int eEF = cplInterface.setMeshEdge(meshID, vE, vF);
+
+        cplInterface.setMeshTriangle(meshID, eBE, eEF, eBF);
+
+      double maxDt = cplInterface.initialize();
+
+      double valueB, valueE, valueF;
+      cplInterface.readScalarData(dataID, vB, valueB);
+      cplInterface.readScalarData(dataID, vE, valueE);
+      cplInterface.readScalarData(dataID, vF, valueF);
+
+      double localCalculatedIntegral = math::geometry::triangleArea(coordTwoB, coordTwoE, coordTwoF) * (valueB + valueE + valueF) / 3.0;
+      utils::MasterSlave::allreduceSum(localCalculatedIntegral, calculatedIntegral, 1);
+
+      cplInterface.advance(maxDt);
+      BOOST_TEST(!cplInterface.isCouplingOngoing(), "Receiving participant should have to advance once!");
+      cplInterface.finalize();
+    }
+    BOOST_TEST(calculatedIntegral == expectedIntegral);
+  }
+}
+*/
```

Comment:
What's the state of this test?

Reply:
Unfortunately, it hangs and results in a timeout since there is missing connectivity information for the received partition. I am not sure whether my setup is correct: Each participant runs on 2 ranks and the mesh is partitioned. Mapping is read mapping.

Reply:
I'll have a look at the test

Reply:
This is the test output when running in isolation: (tidied up)

```
(0) 11:44:13 [partition::ReceivedPartition]:609 in createOwnerInformation:
WARNING: 1 of 8 vertices of mesh MeshOne have been filtered out since they have no influence on the mapping.

(1) 11:44:13 [mapping::NearestProjectionMapping]:160 in computeMapping:
WARNING: 3D Mesh "MeshOne" does not contain triangles. Nearest projection mapping will map to primitives of lower dimension.

(1) 11:44:13 [mapping::Mapping]:89 in scaleConsistentMapping:
ERROR: Connectivity information is missing for the mesh MeshOne. Scaled consistent mapping requires connectivity information.
```

Reply:
The issue seems to be that when running this in parallel, the SolverTwo rank 1 has full overlap with the single triangle of SolverOne rank 1. However, it only shares a single edge with SolverOne rank 0.

During the repartitioning, SolverTwo rank 1 has to compute the mapping for the mesh, received by SolverOne rank 0, which does not contain connectivity information (as expected). This leads to the error.

Reply:
So, too much is filtered out, correct? Sounds like we would need to adjust `tagMeshFirstRound` or `tagMeshSecondRound` of PetRBF.

Reply:
I naively tried to implement extending filtering function such that the connected primitive information won't get lost according to https://github.com/precice/precice/issues/542, however, I did not work on this too much. I can implement this too but I think this PR getting bigger and bigger. Maybe I can create a separate PRs for extending Bounding Box and handling of global indexing of Triangles and Edges?

Reply:
To avoid further confusion, the mentioned test uses nearest-neighbour, not PetRBF.

Reply:
Maybe the duplicate edge leads to problems.
If the nearest-projection mapping chooses to project on a vertex of (SolverOne 0) instead of (SolverOne 1) due to the vertex placement, then it may select as an example 1 vertices from  (SolverOne 0), 2 vertex from (SolverOne 1).
In that case, the triangle is removed.